blueflood-core/src/main/java/com/rackspacecloud/blueflood/inputs/formats/JSONMetricsContainer.java

```diff
@@ -142,19 +146,21 @@ public void setTtlInSeconds(int ttlInSeconds) {
             this.ttlInSeconds = ttlInSeconds;
         }
 
+        public boolean isValid() {
+            // valid if collectionTime is NOT too old AND NOT too far in the future
+            return !tooPast(collectionTime) && !tooFuture(collectionTime);
```

Comment:
Hmmm... This wont work either. Because again, isValid() and getValidationErrors() call these tooPast() and tooFuture() at different times. So a metric can return isValid() true, but got included in the list returned by getValidationErrors(), if its collection time is in the border line of being valid.